packages/host/app/services/recent-files-service.ts

```diff
@@ -47,6 +49,8 @@ export default class RecentFilesService extends Service {
     if (!url) {
       return;
     }
+    // TODO this wont work when visiting files that come from multiple realms in
+    // code mode...
```

Comment:
Good catch. I think the realm url is used here only for showing the realm icon, is that correct @backspace? In that case we can probably just remove `realmURL` from being tracked in `recentFiles`, track only file urls instead, and replace `<RealmInfoProvider @realmURL={{@recentFile.realmURL}}>` with `<RealmInfoProvider @fileURL={{@recentFile.fileURL}}>` in recent-files.gts

Reply:
ya we discussed this [here](https://github.com/cardstack/boxel/pull/662#discussion_r1337249311) but maybe it wouldn’t produce a request every time anymore, as there’s caching?

Reply:
Hmm, it will still produce a request for every file if it hasn't been visited yet (to read the realm info header). Perhaps a more optimized solution would be to keep tracking the realm url but figure it out using the `fetchRealmURL` from `RealmInfoService` instead of reading it from operator mode state service.